File: js-program/child.py
import sys
import json

for line in sys.stdin:
    data = json.loads(line)
    result = {"message": f"Hello, {data['name']}"}
    print(json.dumps(result))
    sys.stdout.flush()
# print(sys.stdin)
# for line in sys.stdin:
#     print(line)

File: js-program/child.js
process.stdin.setEncoding('utf8');

let input = '';
process.stdin.on('data', chunk => {
  input += chunk;
});

process.stdin.on('end', () => {
  try {
    const data = JSON.parse(input);
    const result = { message: `Hello, ${data.name}` };
    process.stdout.write(JSON.stringify(result));
  } catch (err) {
    process.stdout.write(JSON.stringify({ error: 'Invalid JSON' }));
  }
});
